File: workerspayslip.py

import random

# this Creates a list of workers dynamically
def create_workers(num_workers=400):
    workers = []
    for i in range(num_workers):
        worker = {

            # id created
            "id": f"W{i+1}",
            "name": f"Worker_{i+1}",
            "gender": random.choice(["Male", "Female"]),
            "salary": random.randint(5000, 35000),
        }
        workers.append(worker)
    return workers

# then Generate payment slips and conditional statements
def generate_payment_slips(workers):
    for worker in workers:
        try:
            level = "B1"  # level set to default
            salary = worker["salary"]

            # written logic for Employee levels
            if 10000 < salary < 20000:
                level = "A1"
            if 7500 < salary < 30000 and worker["gender"] == "Female":
                level = "A5-F"

            # creativly generating a payment slip
            print(
                f"ID: {worker['id']}, Name: {worker['name']}, "
                f"Gender: {worker['gender']}, Salary: ${salary}, Level: {level}"
            )
        except Exception as e:
            print(f"Error processing worker {worker['id']}: {e}")

# create a main function and catching the error
if __name__ == "__main__":
    try:
        workers_list = create_workers()
        generate_payment_slips(workers_list)
    except Exception as e:
        print(f"Error in main processing: {e}")

File: workerspayslip.R
# Create a list of workers dynamically
create_workers <- function(num_workers = 400) {
  workers <- list()
  for (i in 1:num_workers) {
    worker <- list(
      id = paste0("W", i),
      name = paste0("Worker_", i),
      gender = sample(c("Male", "Female"), 1),
      salary = sample(5000:35000, 1)
    )
    workers[[i]] <- worker
  }
  return(workers)
}

# Generate payment slips and conditional statements
generate_payment_slips <- function(workers) {
  for (worker in workers) {
    tryCatch({
      level <- "B1" # set level to default
      salary <- worker$salary

      # written logic for Employee levels
      if (salary > 10000 && salary < 20000) {
        level <- "A1"
      }
      if (salary > 7500 && salary < 30000 && worker$gender == "Female") {
        level <- "A5-F"
      }

      # creatively generating a payment slip
      cat(
        sprintf(
          "ID: %s, Name: %s, Gender: %s, Salary: $%d, Level: %s\n",
          worker$id, worker$name, worker$gender, salary, level
        )
      )
    }, error = function(e) {
      cat(sprintf("Error processing worker %s: %s\n", worker$id, e))
    })
  }
}

# add a Main function
main <- function() {
  workers_list <- create_workers()
  generate_payment_slips(workers_list)
}

# then run the main function 
main()
